addons/api/addon/generated/models/credential-library.js

```diff
@@ -48,6 +48,7 @@ export default class GeneratedCredentialLibraryModel extends BaseModel {
   //= attributes (vault)
 
   @attr('string', {
+    for: 'vault-generic',
```

Comment:
I ended up using the `for` concept we've talked about before to control whether we serialize the attribute. For some reason however, I couldn't import the type constant and use it, I would always get undefined.

Does anyone know if there's a trick to importing or something I missed?

Reply:
Hmm, I tried this branch locally, removed `for`, and could see the imported type values, and I didn't get `undefined` when I edited a `vault-generic` cred store. Is there a different scenario where you get undefined?

Reply:
Did you mean you replaced the strings with the constants and were able to save?

I was trying 
```
import {
  TYPE_CREDENTIAL_LIBRARY_VAULT_GENERIC,
  TYPE_CREDENTIAL_LIBRARY_VAULT_SSH_CERTIFICATE,
} from '../models/credential-library';
```

and

```
import {
  TYPE_CREDENTIAL_LIBRARY_VAULT_GENERIC,
  TYPE_CREDENTIAL_LIBRARY_VAULT_SSH_CERTIFICATE,
} from 'api/models/credential-library';
```

And I wasn't getting a value in the `for` option when serializing while using these constants

Reply:
Ahh, I misinterpreted the original comment. I get `undefined` as well; however, if I move the constants to `baseModel`, I can access them just fine by importing from there, and I can also access constants from other models, such as targets.  🤔

Reply:
Fine for now.  In the future I think types will live in a separate place from models.